WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -47,8 +49,26 @@ public class ProductsScreen: ScreenObject {
     }
 
     @discardableResult
-    public func selectProduct(atIndex index: Int) throws -> SingleProductScreen {
-        XCUIApplication().tables.cells.element(boundBy: index).tap()
+    public func selectProductByIndex(atIndex index: Int) throws -> SingleProductScreen {
+        app.tables.cells.element(boundBy: index).tap()
         return try SingleProductScreen()
     }
+
+    public func selectProductByName(name: String) throws -> SingleProductScreen {
+        app.tables.cells.staticTexts[name].tap()
+        return try SingleProductScreen()
+    }
+
+    public func verifyProductOnProductsScreen(count: Int, name: String, status: String) throws -> ProductsScreen {
+        XCTAssertTrue(try app.getTextVisibilityCount(text: name) == 1, "Product name does not exist!")
+        XCTAssertTrue(app.tables.cells.count == count, "Expecting \(count) products, got \(app.tables.cells.count) instead!")
+        XCTAssertTrue(try app.getTextVisibilityCount(text: status) > 0, "Stock status does not exist!")
```

Comment:
> For now, to make it somewhat meaningful, I'll validate the count matches what's expected from the mock response (we still have control over this) and add a TODO to update this for when accessibilityIdentifier is added.

I think this is a good compromise for now. Something to keep in mind for future (maybe very distant): while we have control of the mocks now, this might change sometime (e.g. we'll switch to real API), and because of that I think it's important to keep the tests and actual test data as isolated as possible. Asserting functions should just accept the data (no matter if it's coming from mocks, or an array created right in the test file) and correctly handle the cases when test data size changes.